.github/workflows/typespec-migration-validation.yaml

```diff
@@ -0,0 +1,24 @@
+name: TypeSpec Migration Validation
+
+on: pull_request
+
+permissions:
+  contents: read
+
+jobs:
+  typespec-validation:
+    name: TypeSpec Validation
+    runs-on: ubuntu-24.04
+
+    steps:
```

Comment:
I think you can add a first step, which uses github-script to call a REST API to list the labels on the issue, and no-op if you don't have a `typespec-conversion-*` label.